packages/cli/src/workflows/workflows.controller.ee.ts

```diff
@@ -58,3 +58,37 @@ EEWorkflowController.put('/:workflowId/share', async (req: WorkflowRequest.Share
 
 	return res.status(200).send();
 });
+
+EEWorkflowController.get(
+	'/:id',
+	(req: WorkflowRequest.Get, res, next) => (req.params.id === 'new' ? next('router') : next()), // skip ee router and use free one for naming
+	ResponseHelper.send(async (req: WorkflowRequest.Get) => {
+		const { id: workflowId } = req.params;
+
+		if (Number.isNaN(Number(workflowId))) {
+			throw new ResponseHelper.ResponseError(`Workflow ID must be a number.`, undefined, 400);
+		}
+
+		const workflow = await EEWorkflows.get(
+			{ id: parseInt(workflowId, 10) },
+			{ relations: ['shared', 'shared.user', 'shared.role'] },
+		);
+
+		if (!workflow) {
+			throw new ResponseHelper.ResponseError(
+				`Workflow with ID "${workflowId}" could not be found.`,
+				undefined,
+				404,
+			);
+		}
+
+		const userSharing = workflow.shared?.find((shared) => shared.user.id === req.user.id);
+
+		if (!userSharing && req.user.globalRole.name !== 'owner') {
```

Comment:
I think we should extract the `globalRole` check into a middleware. This won't be the last place we need this check.